client/my-sites/checkout/composite-checkout/components/wp-order-review-line-items.tsx

```diff
@@ -774,10 +774,15 @@ function DomainDiscountCallout( {
 	const translate = useTranslate();
 
 	const isFreeBundledDomainRegistration = product.is_bundled && product.item_subtotal_integer === 0;
-	if ( isFreeBundledDomainRegistration && ! hasPlanInCart ) {
+
+	if ( isFreeBundledDomainRegistration && product.is_renewal ) {
 		return <DiscountCallout>{ translate( 'Domain credit applied' ) }</DiscountCallout>;
 	}
 
+	if ( isFreeBundledDomainRegistration && ! hasPlanInCart ) {
```

Comment:
Wouldn't we shod this message when there *is* a plan in your cart?

Reply:
No, it follows the logic of the popover cart (on the domains page). If you already have a plan and buy a new domain, it's saying "this is covered by your plan". It's just a friendlier version of "domain credit applied" for that situation.